### PR TITLE
added validation messages to sfi23 statement scheme and adjusted allo…

### PR DESCRIPTION
**Pull request by JungaBunga**
…wed amount of characters under year field to prevent year to be more than 4 characters

### CODE REVIEW

**Review by samplackett — DISMISSED** [review]
Approved, however a general question...

I like how some of the errors use the variable themselves, ie min and max FRN. Is there any value to using all of our number4, number5 variables within the error themselves too, rather than hard coding the number?

Don't think it's a blocker to this merge, but perhaps one to think about if that makes sense to do before we complete this one.

File: app/messaging/schemas/sfi-23-quarterly-statement.js [hunk 1]
```diff
@@ -37,223 +45,216 @@ const addressSchema = Joi.object({
...
 })
 
 module.exports = Joi.object({
   address: addressSchema,
-  businessName: Joi.string().max(number100).required().messages({
+  businessName: Joi.string().max(maxSchemeNameLength).required().messages({
     'string.base': 'Business name must be a string',
-    'string.max': 'Business name must be at most 100 characters',
-    'any.required': 'Business name is not present but it is required'
+    'string.max': `Business name must be at most ${maxSchemeNameLength} characters`,
```
Comment (samplackett) [inline app/messaging/schemas/sfi-23-quarterly-statement.js:156]: think we probably want this to be maxBusinessNameLength

File: app/messaging/schemas/sfi-23-quarterly-statement.js [hunk 2]
```diff
@@ -37,223 +45,216 @@ const addressSchema = Joi.object({
...
     'number.integer': 'Calculation ID must be an integer',
-    'any.required': 'Calculation ID is not present but it is required'
+    'any.required': 'Calculation ID is required'
   }),
-  paymentPeriod: Joi.string().max(number200).allow('', null).optional().messages({
+  paymentPeriod: Joi.string().max(maxPaymentPeriodLength).required().messages({
     'string.base': 'Payment period must be a string',
-    'string.max': 'Payment period must be at most 200 characters'
-  }),
-  paymentAmount: Joi.number().required().messages({
```
Comment (samplackett) [inline app/messaging/schemas/sfi-23-quarterly-statement.js]: I could be missing some info but is there a reason we've now removed paymentAmount and transactionDate?

Comment (JungaBunga) [reply app/messaging/schemas/sfi-23-quarterly-statement.js]: omg, I am so done with copilot removing fields for no reason, it's bloody annoying

Comment (samplackett) [reply app/messaging/schemas/sfi-23-quarterly-statement.js]: at least we won't be fully replaced by AI for a couple of years...

**Review by samplackett — APPROVED** [review]
abdc

**Pull request merged**